Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/GeometricFillTool.java

```diff
@@ -142,6 +152,18 @@ private void createAndSetBitmap() {
 			drawPaint.setShader(checkeredPattern.getShader());
 		}
 
+		if (drawPaint.getStyle() == Style.STROKE) {
```

Comment:
Take a look at [Rect.inset(int, int)](https://developer.android.com/reference/android/graphics/Rect#inset(int,%20int)), this could improve the readability of your code.

Reply:
I am not sure inset would improve readability. shapeRect still has to be initialized once for inset to work, and since fill and outline have different calculations for shapeRect, everytime the fill/outline option is changed, shapeRect would need a new calculation before inset can be used again.